internal/tray-agent/agent/client/crdClients.go

```diff
@@ -35,28 +39,29 @@ type crdManager struct {
 //required CRD.
 func (ctrl *AgentController) initCRDManager() error {
 	//struct init
-	mngr := &crdManager{clientMap: make(map[CustomResource]*CRDController)}
-	ctrl.crdManager = mngr
+	manager := &crdManager{clientMap: make(map[CustomResource]*CRDController)}
+	ctrl.crdManager = manager
 	kubeconfig, set := os.LookupEnv(EnvLiqoKConfig)
 	if !set {
 		return errors.New("no kubeconfig provided")
 	}
 	//creation of each single CRDController
 	var err error
 	var crdCtrl *CRDController
-	// - CLUSTERCONFIG
-	crdCtrl, err = createClusterConfigController(kubeconfig)
-	if err != nil {
-		return errors.New("connection error on clusterconfigs client creation")
+	for _, cr := range customResources {
```

Comment:
Why are you using this loop that iterates over a slice and switches on the elements? Can't you simply call the functions one after the other?

Reply:
I thought was a good idea to reuse the common code for each 'creation', like (60-63), instead of always rewriting the same checks.

Reply:
I think it is a bit confusing, it seems we are adding complexity where the workflow is pretty simple.

Reply:
I re-checked the code: I moved to the iteration mechanism also to use the same scheme of the tests. 

The *customResources* variable, similarly to notifyChannelNames, is a slice containing all the CRDs currently managed by the AgentController and the *TestAgentControllerComponentsReadiness* iterates on that, checking the respective clients are active. I thought that using the same approach in this init phase could have compensated a mechanism maybe a little convoluted.

If you think it's still worth reverting to sequential calls, I can do it without problems

Reply:
yes please, try using the clearest and simplest approach if there is no complexity.